AWSIoTTests/AWSIoTDataManagerTests.swift

```diff
@@ -741,18 +741,35 @@ class AWSIoTDataManagerTests: XCTestCase {
         let testMessage = "Test Message"
         let testTopic = "TestTopic"
         gotMessage.expectedFulfillmentCount = 5
+
+        let subAckExpectation = self.expectation(description: "Subscription should be acknowledged")
+        let subAckCallback: AWSIoTMQTTAckBlock = {
+            subAckExpectation.fulfill()
+        }
+
         //Subscribe to TestTopic
-        iotDataManager.subscribe(toTopic: testTopic, qoS: .messageDeliveryAttemptedAtLeastOnce, messageCallback: {
-            (payload) ->Void in
-            let stringValue:String = NSString(data: payload, encoding: String.Encoding.utf8.rawValue)! as String
-            print("received: \(stringValue)")
-            XCTAssertEqual(testMessage, stringValue)
-            gotMessage.fulfill()
-        })
+        let subStatus = iotDataManager.subscribe(toTopic: testTopic,
+                                                 qoS: .messageDeliveryAttemptedAtLeastOnce,
+                                                 messageCallback: {
+                                                    (payload) ->Void in
+                                                    let stringValue:String = NSString(data: payload, encoding: String.Encoding.utf8.rawValue)! as String
+                                                    print("received: \(stringValue)")
+                                                    XCTAssertEqual(testMessage, stringValue)
+                                                    gotMessage.fulfill()
```

Comment:
Hmmm If we publish 5 times, does that mean this callback will get called 5 times? should we have some logic which checks to make sure this callback gets called 5 times?

Reply:
We are already checking it with `gotMessage.expectedFulfillmentCount = 5`

Reply:
I *think* IoT is set up to use MQTT QoS 1 by default, which means messages will be delivered "at least once". But, AWS IoT may send a message multiple times [**even when specifying QoS 0**](https://docs.aws.amazon.com/iot/latest/developerguide/mqtt.html)).

That means we may receive the callback **more** than 5 times depending on network conditions. So, keep `gotMessage.expectedFulfillmentCount = 5`, but also make sure that `gotMessage.assertForOverfulfill = false`. Extend that to all of our message expectations: we need to assert the minimum number of fulfillment calls, but not assert on overfulfill.

In future PubSub work, we should manage that on behalf of the customer by keeping track of delivery by message ID, so that we can remove QoS considerations and deliver to the client "exactly once" even though the server may send multiple times.

Reply:
Updated the pr